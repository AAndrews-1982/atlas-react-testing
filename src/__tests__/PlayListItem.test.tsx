// Import necessary modules and components
import { render } from "@testing-library/react";
import { PlayListItem } from "../components/PlayListItem";
import { AppContext } from "../components/AppContext";
import { expect, it, vi } from "vitest";

// Define mock data for the context
const mockSongs = [
  { title: "Test Song 1", artist: "Test Artist 1", duration: "3:00", cover: "cover1.png" },
  { title: "Test Song 2", artist: "Test Artist 2", duration: "4:00", cover: "cover2.png" }
];

const mockContextValue = {
  songs: mockSongs,
  currentSong: 0,
  setCurrentSong: vi.fn()
};

it("should render the PlayListItem component and match the snapshot", () => {
  const { asFragment } = render(
    <AppContext.Provider value={mockContextValue}>
      <PlayListItem songTitle={mockSongs[0].title} artist={mockSongs[0].artist} playTime={mockSongs[0].duration} />
    </AppContext.Provider>
  );

  // Verify that the rendered output matches the snapshot
  expect(asFragment()).toMatchSnapshot();
});
