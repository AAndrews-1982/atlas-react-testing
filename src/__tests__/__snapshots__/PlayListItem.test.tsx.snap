// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the PlayListItem component and match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="playlist-item mb-1 flex h-10 cursor-pointer items-center justify-between rounded-md text-sm font-medium bg-hover dark:bg-dark-hover active:bg-active dark:active:bg-dark-active"
  >
    <div
      class="song-details"
    >
      <div
        class="song-title text-sm font-bold mb-0"
      >
        Test Song 1
      </div>
      <div
        class="artist-name text-sm text-muted-text dark:text-dark-muted-text"
      >
        Test Artist 1
      </div>
    </div>
    <div
      class="play-time text-sm text-muted-text dark:text-dark-muted-text"
    >
      3:00
    </div>
  </div>
</DocumentFragment>
`;
